tests/components/gios/test_config_flow.py

```diff
@@ -1,138 +1,104 @@
 """Define tests for the GIOS config flow."""
 
-import json
-from unittest.mock import patch
+from unittest.mock import MagicMock
 
-from gios import ApiError
+from gios import ApiError, InvalidSensorsDataError
+import pytest
 
-from homeassistant.components.gios import config_flow
 from homeassistant.components.gios.const import CONF_STATION_ID, DOMAIN
 from homeassistant.config_entries import SOURCE_USER
 from homeassistant.const import CONF_NAME
 from homeassistant.core import HomeAssistant
 from homeassistant.data_entry_flow import FlowResultType
 
-from . import STATIONS
-
-from tests.common import async_load_fixture
-
 CONFIG = {
     CONF_NAME: "Foo",
     CONF_STATION_ID: "123",
 }
 
+pytestmark = pytest.mark.usefixtures("mock_gios")
+
 
 async def test_show_form(hass: HomeAssistant) -> None:
     """Test that the form is served with no input."""
-    with patch(
-        "homeassistant.components.gios.coordinator.Gios._get_stations",
-        return_value=STATIONS,
-    ):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}
-        )
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
 
     assert result["type"] is FlowResultType.FORM
     assert result["step_id"] == "user"
+    assert len(result["data_schema"].schema[CONF_STATION_ID].config["options"]) == 2
 
 
-async def test_form_with_api_error(hass: HomeAssistant) -> None:
+async def test_form_with_api_error(hass: HomeAssistant, mock_gios: MagicMock) -> None:
     """Test the form is aborted because of API error."""
-    with patch(
-        "homeassistant.components.gios.coordinator.Gios._get_stations",
-        side_effect=ApiError("error"),
-    ):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}
-        )
+    mock_gios.create.side_effect = ApiError("error")
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
 
     assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "cannot_connect"
 
 
-async def test_invalid_sensor_data(hass: HomeAssistant) -> None:
+async def test_invalid_sensor_data(hass: HomeAssistant, mock_gios: MagicMock) -> None:
```

Comment:
`test_invalid_sensor_data` and `test_cannot_connect` are almost the same. We can combine them into a single test and use `parameterize`.

Reply:
Done. We use `parameterize` now.